src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -0,0 +1,182 @@
+        return
+    return accounts[0].object_id
+
+def _get_object_id_by_upn(graph_client, upn):
+    accounts = list(graph_client.users.list(filter="userPrincipalName eq '{}'".format(upn)))
+    if not accounts:
+        logger.warning("Unable to find user with upn '%s'", upn)
+        return
+    if len(accounts) > 1:
+        logger.warning("Multiple users principals found with upn '%s'. "\
+                    "You can avoid this by specifying object id.", upn)
+        return
+    return accounts[0].object_id
+
+def _get_object_id_from_subscription(graph_client, subscription):
+    if subscription['user']:
+        if subscription['user']['type'] == 'user':
+            return _get_object_id_by_upn(graph_client, subscription['user']['name'])
+        elif subscription['user']['type'] == 'servicePrincipal':
+            return _get_object_id_by_spn(graph_client, subscription['user']['name'])
+        else:
+            logger.warning("Unknown user type '%s'", subscription['user']['type'])
+    else:
+        logger.warning('Current credentials are not from a user or service principal. '\
+                    'Azure Key Vault does not work with certificate credentials.')
+
+def _get_object_id(graph_client, subscription=None, spn=None, upn=None):
+    if spn:
+        return _get_object_id_by_spn(graph_client, spn)
+    if upn:
+        return _get_object_id_by_upn(graph_client, upn)
+    return _get_object_id_from_subscription(graph_client, subscription)
+
+def create_keyvault(client, resource_group_name, vault_name, location, #pylint:disable=too-many-arguments
+                    sku=SkuName.standard.value,
+                    enabled_for_deployment=None,
+                    enabled_for_disk_encryption=None,
+                    enabled_for_template_deployment=None,
+                    no_self_perms=False,
+                    tags=None):
+    from azure.cli._profile import Profile
+    profile = Profile()
+    cred, _, tenant_id = profile.get_login_credentials(for_graph_client=True)
+    graph_client = GraphRbacManagementClient(cred, tenant_id)
+    subscription = profile.get_default_subscription()
+    if no_self_perms:
+        access_policies = []
+    else:
+        # TODO Use the enums instead of strings when new keyvault SDK is released
+        # https://github.com/Azure/azure-sdk-for-python/blob/dev/azure-mgmt-keyvault/
+        # azure/mgmt/keyvault/models/key_vault_management_client_enums.py
+        permissions = Permissions(keys=['get',
+                                        'create',
+                                        'delete',
+                                        'list',
+                                        'update',
+                                        'import',
+                                        'backup',
+                                        'restore'],
+                                  secrets=['all'])
+        object_id = _get_current_user_object_id(graph_client)
```

Comment:
Just a question, if you currently log in as a service principal, does this call work?